autoload/lsc/cursor.vim

```diff
@@ -26,22 +26,76 @@ function! lsc#cursor#enableReferenceHighlights(filetype)
   let s:highlight_support[a:filetype] = v:true
 endfunction
 
+" Given a single-line string, returns a potentially multi-line string that
+" fits within the given maximum width.
+" Adapted from https://vi.stackexchange.com/a/4930
+function! WrapText(text, width)
+  let l:height = 1
+  let l:line = ''
+  let l:ret  = ''
+  for word in split(a:text)
+    if len(l:line) + len(word) + 1 > a:width
+      if len(l:ret)
+        let l:ret .= "\n"
+        let l:height += 1
+      endif
+      let l:ret .=  l:line
+      let l:line = ''
+    endif
+    if len (l:line)
+      let l:line .= ' '
+    endif
+    let l:line .= strtrans(word)
+  endfor
+  let l:ret .= "\n" . l:line
+  return {"message": l:ret, "height": l:height}
```

Comment:
I have a WIP refactoring that removes the dictionary which simplifies the logic quite a bit. I also would like to make the functions local. I've never written vimscript before so it was all new to me 😅